test/services/news/index.test.js

```diff
@@ -458,6 +492,276 @@ describe('news service', () => {
+				await News.create([
+					{
+						schoolId,
+						creatorId: (await createTestUser())._id,
+						title: 'school A news',
+						content: 'some string',
+						displayAt: Date.now(),
+					},
+					{
+						schoolId,
+						creatorId: (await createTestUser())._id,
+						title: 'Unpublished newstitle',
+						content: 'Who know what the future holds',
+						displayAt: now + 20000,
+					},
+					{
+						schoolId,
+						creatorId: (await createTestUser())._id,
+						title: 'Unpublished newstitle (2)',
+						content: 'Unkown event about to happen',
+						displayAt: new Date(now + 20000),
+					},
+					{
+						schoolId,
+						creatorId: (await createTestUser())._id,
+						title: 'school A news (2)',
+						content: 'some string',
+						displayAt: new Date('2019-09-01T10:01:58.135Z'),
+					},
+				]);
+				const user = await createTestUser({
+					schoolId,
+					roles: 'student',
+				});
+				const params = await generateRequestParamsFromUser(user);
+				// params.query = { pusblished: true }; this is deafault
+				const result = await newsService.find(params);
+				expect(result.total).to.equal(2);
+				expect(
+					result.data.every((item) => item.title.includes('school A news')),
+				).to.equal(true);
+				expect(
+					result.data.every((item) => item.title.includes('Unpublished newstitle')),
+				).to.equal(false);
+			});
+
+			it('should only return published team news', async () => {
+				const schoolId = (await createTestSchool())._id;
+				const teams = teamHelper(app, { schoolId });
+				const user = await createTestUser({
+					schoolId,
+					roles: 'administrator',
+				});
+				const user2 = await createTestUser({
+					schoolId,
+					roles: 'teacher',
+				});
+				const teamA = await teams.create(user);
+				const teamB = await teams.create(user2);
+
+				const now = Date.now();
+				await News.create([
+					{
+						schoolId,
+						creatorId: (await createTestUser())._id,
+						title: 'school news',
+						content: 'this is the content',
+						displayAt: now + 20000,
+					},
+					{
+						schoolId,
+						creatorId: user._id,
+						title: 'team A news',
+						content: 'even more content',
+						target: teamA._id,
+						targetModel: 'teams',
+						displayAt: new Date(now + 20000),
+					},
+					{
+						schoolId: (await createTestSchool())._id, // team news created at another school
+						creatorId: (await createTestUser())._id,
+						title: 'team A news 2',
+						content: 'even more content',
+						target: teamA._id,
+						targetModel: 'teams',
+						displayAt: now,
+					},
+					{
+						schoolId,
+						creatorId: (await createTestUser())._id,
+						title: 'team B news',
+						content: 'we have content, too',
+						target: teamB._id,
+						targetModel: 'teams',
+						displayAt: new Date('2019-09-01T10:01:58.135Z'),
+					},
+				]);
+				const params = await generateRequestParamsFromUser(user);
+				const result = await newsService.find(params);
+				expect(result.total).to.equal(1);
```

Comment:
the total test is very important to check that there are no any other news from a different scope are shown here, maybe for that reason in the beginning create one news item, that is placed in a different team, school, and so on...

Reply:
There are two teams [teamA,teamB] in the function to test that. Also a newselement without team to see if it interfere in anyway.